src/command.c

```diff
@@ -465,6 +469,78 @@ int cmd_proxy(struct command *cmd, struct redis_data *data)
     return CORVUS_OK;
 }
 
+int cmd_config(struct command *cmd, struct redis_data *data)
+{
+    ASSERT_TYPE(data, REP_ARRAY);
+    ASSERT_ELEMENTS(data->elements >= 2, data);
+
+    struct redis_data *op = &data->element[1];
+    struct redis_data *opt = &data->element[2];
+    ASSERT_TYPE(op, REP_STRING);
+    ASSERT_TYPE(opt, REP_STRING);
+
+    char type[op->pos.str_len + 1];
+    char option[opt->pos.str_len + 1];
+    if (pos_to_str(&op->pos, type) == CORVUS_ERR
+            || pos_to_str(&opt->pos, option) == CORVUS_ERR) {
+        LOG(ERROR, "cmd_config: parse error");
+        return CORVUS_ERR;
+    }
+    if (strcasecmp(type, "SET") == 0) {
+        if (strcasecmp(option, "NODE") == 0) {
+            // config set node host:port,host1:port1
+            if (data->elements != 4) {
+                cmd_mark_fail(cmd, rep_config_parse_err);
+            } else {
+                char value[data->element[3].pos.str_len + 1];
+                // the host-port pair is in form "1.1.1.1:1"(9 bytes) at least
+                if (data->element[3].pos.str_len < 9
+                        || pos_to_str(&data->element[3].pos, value) != CORVUS_OK) {
+                    cmd_mark_fail(cmd, rep_config_parse_err);
+                } else if (config_add("node", value) != CORVUS_OK) {
+                    cmd_mark_fail(cmd, rep_config_addr_err);
+                } else {
+                    slot_create_job(SLOT_UPDATE);
+                    conn_add_data(cmd->client, (uint8_t*) rep_ok, strlen(rep_ok),
+                            &cmd->rep_buf[0], &cmd->rep_buf[1]);
+                    CMD_INCREF(cmd);
+                    cmd_mark_done(cmd);
+                }
+            }
+        } else {
+            cmd_mark_fail(cmd, rep_addr_err);
+        }
+    } else if (strcasecmp(type, "GET") == 0) {
+        if (strcasecmp(option, "NODE") == 0) {
```

Comment:
aha, sorry about changes after merging.
I change the assert for elements judging ensure that.